### ISSUE TITLE
Rethink storage account naming convention

### ISSUE THREAD

**Issue by SvenAelterman** [dcampman/Azure-SRE#2]
The existing naming convention doesn't tend to work well with storage accounts (and Key Vaults) due to those resource types' names' low max length.
For the storage account, we might just need to use `st` + random digits.

**Comment by dcampman** [comment]
Removed the naming convention variable, added a guid function built from the baseName variable, and now requiring the subworkload parameter.

var stgNameClean = replace(take(guid(subscription().id, resourceGroup().id, baseName), 22), '-', '')

The value passed to the guid function will be different for every module call when the now mandatory sub-workload parameter is sent. I also included the Subscription Id and Resource Group Id to give it a little more singularity. Using the guid function is required to keep a consistent deployment on subsequent runs but has limitations due to the parameters being used to generate the random guid.

If we use newGuid, which would generate a new guid ever time the deployment runs, it would create a new storage account every time, instead of updating the previous deployment.
